alacritty/src/daemon.rs

```diff
@@ -43,18 +49,44 @@ where
         .map(|_| ())
 }
 
+// get working directory of controlling process
+#[cfg(not(windows))]
+pub fn foreground_process_path() -> Result<PathBuf, Box<dyn Error>> {
+    let mut pid = unsafe { libc::tcgetpgrp(tty::master_fd()) };
+    if pid < 0 {
+        pid = tty::child_pid();
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "freebsd")))]
+    let link_path = format!("/proc/{}/cwd", pid);
+    #[cfg(target_os = "freebsd")]
+    let link_path = format!("/compat/linux/proc/{}/cwd", pid);
+
+    #[cfg(not(target_os = "macos"))]
+    let cwd = fs::read_link(link_path)?;
+
+    // this might not be a io::Result, so a Boxed dyn Error is needed
```

Comment:
You don't need to explain type signatures in your comments.

Reply:
Ok, I removed it. I felt like commenting it because at first glance (without looking into the macos part) it might seem like an `io::Result` would be the right choice.